src/core/load-fragment.c

```diff
@@ -1338,6 +1348,22 @@ int config_parse_exec_cpu_affinity(const char *unit,
         assert(rvalue);
         assert(data);
 
+        if (streq(rvalue, "NUMA")) {
+                if (c->numa_policy.nodes.set) {
+                        int r;
+
+                        r = numa_to_cpu_set(&c->numa_policy, &c->cpu_set);
+                        if (r < 0)
+                                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to derive CPU affinity mask from NUMA node mask, ignoring");
+                        return 0;
+                } else
+                        /* We haven't seen NUMAMask= yet, let's remember we need to set CPUAffinity= once we have parsed it */
+                        c->cpu_affinity_numa = true;
+
+
+        } else if (isempty(rvalue))
+                c->cpu_affinity_numa = false;
```

Comment:
It'd be nice to make the "numa" comparison case-insensitive. "NUMA" feels a bit awkward.

I don't like the semantics of the assignment here. Settings should be configurable in any order, so the result should be the same if I do the assignments to CPUAffinity= and NUMAMask= in whatever order, or possibly interleaved. This means that the `c->cpu_affinity_numa` should only be evaluated at the very end, after all parsing is done.

I see two possibilities when explicit ranges and "numa" is mixed:
1. setting CPUAffinity=numa overrides previous settings, so `c->numa_policy.nodes` is emptied when this setting is encountered, and vice versa, when a different `CPUAffinity=…` assignment happens, `c->cpu_affinity_numa` is unset, and the empty assignment `CPUAffinity=` happens, it both empties `c->numa_policy.nodes` and unsets `c->cpu_affinity_numa`.
2. CPUAffinity=numa *adds* to the other CPUAffinity= assignments, so it only sets `c->cpu_affinity_numa`, and other assignments only set `.nodes`, but the empty assignment still unsets both.

Which is better depends on how people would use this: i.e. if it is useful to set the CPUAffinity to numa + some cpus, or would one always simply set it to either the numa list or a static list. Either way, this should be mentioned in the documentation.